thorlcr/activities/indexwrite/thindexwrite.cpp

```diff
@@ -280,6 +280,10 @@ class IndexWriteActivityMaster : public CMasterActivity
                 bloom->setProp("@bloomProbability", pval.str());
             }
             container.queryTempHandler()->registerFile(fileName, container.queryOwner().queryGraphId(), 0, false, WUFileStandard, &clusters);
+            props.setPropInt64("@timeDiskWriteIO", statsCollection.getStatisticSum(StTimeDiskWriteIO));
```

Comment:
I don't think you want all of these recorded in the file information - by using the merge function they will be available in the activity stats, and that should be sufficient.  @jakesmith what do you think is useful to store?

Reply:
right, yes all will be at the activity level, but we want to propagate some to retain at a logical file level.

>Which should publish more than one disk write statistic (e.g., time, total written) to the activity stats.

@ghalliday - was you previous comment re. ensuring published (the index stats.) at activity level, but not propagating them to the logical file properties?

Reflecting on that (what's wanted at logical file level).
StSizeDiskWrite - should be very similar to what is already accounted for and published by the write activities in their @size and @compressedSize properties, so I don't think to add this.
StNumDiskRetries - I can't see anywhere that updates that statistic, so remove for now?

StTimeDiskWriteIO - not sure needed at logical file level, likely says more about the contention at the time, than anything else, although we could add for now, as cheap/free, and see if useful later.
NumDiskWrites - we do want this one.

Reply:
Yes, I should have been clearer.

My comment from before was meant to be that if you use the mergeStatistic call then all the disk stats would automatically be published in the *activity* (fixing a bug in the process).  I would only publish to the file stats that aren't there already and are needed for the costs.  I think that is only the number of writes (as you were doing already).